hack/ensure_job_cluster.py

```diff
@@ -0,0 +1,98 @@
+#!/usr/bin/env python3
+"""This script ensures the cluster of prowjobs is defined as expected.
+e.g., python ./hack/ensure_job_cluster.py
+"""
+
+import argparse
+import logging
+import os
+import yaml
+
+
+def get_desired_cluster(file_path, job):
+    """get the desired cluster for a given job defined in file_path."""
+    if job.get("agent", "") != "kubernetes":
+        return ""
+    if job["name"] in JOB_MAP:
+        return JOB_MAP[job["name"]]
+    if job["name"].endswith("-build01") or migrated(file_path):
+        return BUILD01_CLUSTER
+    return ""
+
+
+def get_updated_jobs(file_path, jobs, overwrite):
+    """get jobs with correct cluster."""
+    new_jobs = []
+    for job in jobs:
+        cluster = get_desired_cluster(file_path, job)
+        if cluster != job.get("cluster", ""):
+            message = "expected the cluster of job {} to be {}, but found {}"\
+                .format(job["name"], cluster, job.get("cluster", ""))
+            if not overwrite:
+                raise Exception(message)
+            logging.warning(message)
+            job["cluster"] = cluster
+        new_jobs.append(job)
+    return new_jobs
+
+
+def migrated(file_path):
+    """check if the jobs defined in file_path are migrated."""
+    # we do not migrate the periodis in release repo
+    # due to https://github.com/openshift/release/pull/7178
+    if file_path.endswith('periodics.yaml') and 'openshift/release/' in file_path:
+        return False
+    return False
+
+def ensure(job_dir, overwrite):
+    """ensure prow jobs' cluster."""
+    i = 0
+    for dirpath, _, filenames in os.walk(job_dir):
+        for filename in filenames:
+            if filename.endswith('.yaml'):
+                file_path = os.path.join(dirpath, filename)
+                i += 1
+                if i % 100 == 1:
```

Comment:
Why?

Reply:
Removed.

Tradeoff between no logs to show progress and too many logs to see useful ones.